core/src/main/java/edu/wpi/grip/core/events/SourceStartedEvent.java

```diff
@@ -0,0 +1,19 @@
+package edu.wpi.grip.core.events;
+
+
+import edu.wpi.grip.core.Source;
+
+/**
+ * An event that occurs when a source is started.
+ */
+public class SourceStartedEvent {
+    private final Source source;
+
+    public SourceStartedEvent(Source source){
```

Comment:
If you're going to say something is bad can you need to provide an alternative. I don't know how to do it any differently.

I did it this way because in case a source crashes then the UI will know to toggle the button back to showing the start button again.